core/src/main/web/app/mainapp/services/sessionmanager.js

```diff
@@ -892,8 +893,11 @@
         }
       },
       backup: function() {
-        if (_sessionId && !_notebookModel.isEmpty()) {
-          return bkSession.backup(_sessionId, generateBackupData());
+        if (_sessionId && !_notebookModel.isEmpty() && _needsBackup) {
+          return bkSession.backup(_sessionId, generateBackupData())
+          .then(function() {
+            _needsBackup = false;
```

Comment:
Here I think you should clear _needsBackup before executing bkSession.backup() - what happens if I'm editing the notebook and I put edits during the time it takes for the request to go out to when the promise is fulfilled?